File: TETRIS_AI4/src/main/scala/nu/tengstrand/tetrisanalyzer/piece/PieceT.scala
package nu.tengstrand.tetrisanalyzer.piece

object PieceT {
  def apply() = new PieceT
}

class PieceT extends Piece {
  val number = 6.toByte
  val character = 'T'
  protected val widths = Array(3, 2, 3, 2)
  protected val heights = Array(2, 3, 2, 3)
  protected val shapes = Array(
    new PieceShape(Array(Point(0,0), Point(1,0), Point(2,0), Point(1,1))),
    new PieceShape(Array(Point(0,0), Point(0,1), Point(1,1), Point(0,2))),
    new PieceShape(Array(Point(1,0), Point(0,1), Point(1,1), Point(2,1))),
    new PieceShape(Array(Point(1,0), Point(0,1), Point(1,1), Point(1,2)))
  )
}

File: TetrisAI_3/src/com/github/tetrisanalyzer/piece/PieceT.java
package com.github.tetrisanalyzer.piece;

import com.github.tetrisanalyzer.settings.PieceSettings;

public class PieceT extends Piece {

    PieceT(PieceSettings settings) {
        super(settings);
    }

    @Override public byte number() { return 7; }
    @Override public char character() { return 'T'; }
    @Override protected int[] widths() { return new int[] { 3, 2, 3, 2 }; }
    @Override protected int[] heights() { return new int[] { 2, 3, 2, 3 }; }
    @Override protected PieceShape[] shapes() {
        return new PieceShape[] {
            new PieceShape(new Point(0,0), new Point(1,0), new Point(2,0), new Point(1,1)),
            new PieceShape(new Point(0,0), new Point(0,1), new Point(1,1), new Point(0,2)),
            new PieceShape(new Point(1,0), new Point(0,1), new Point(1,1), new Point(2,1)),
            new PieceShape(new Point(1,0), new Point(0,1), new Point(1,1), new Point(1,2))
        };
    }
}
